tests/unit/test_characterProcessing.py

```diff
@@ -0,0 +1,308 @@
+		triple="1.11"),
+	NrProcTestCase(origin="1.111",
+		single="1.1  1  1",
+		double="1.1  11",
+		triple="1.111"),
+	NrProcTestCase(origin="1.1111",
+		single="1.1  1  1  1",
+		double="1.11  11",
+		triple="1.1  111"),
+	NrProcTestCase(origin="11.111",
+		single="1  1.1  1  1",
+		double="11.1  11",
+		triple="11.111"),
+	NrProcTestCase(origin="11.1111",
+		single="1  1.1  1  1  1",
+		double="11.11  11",
+		triple="11.1  111"),
+	NrProcTestCase(origin="111.111",
+		single="1  1  1.1  1  1",
+		double="1  11.1  11",
+		triple="111.111"),
+	NrProcTestCase(origin="1111.1111",
+		single="1  1  1  1.1  1  1  1",
+		double="11  11.11  11",
+		triple="1  111.1  111"),
+	# Decimal with letters
+	NrProcTestCase(origin="1a.234b",
+		single="1a.2  3  4b",
+		double="1a.2  34b",
+		triple="1a.234b"),
+	NrProcTestCase(origin="1a.2.345b",
+		single="1a.2.3  4  5b",
+		double="1a.2.3  45b",
+		triple="1a.2.345b"),
+	NrProcTestCase(origin="1a.2.3456b",
+		single="1a.2.3  4  5  6b",
+		double="1a.2.34  56b",
+		triple="1a.2.3  456b"),
+]
+
+decimalComma = [
+	NrProcTestCase(origin="1,1",
+		single="1,1",
+		double="1,1",
+		triple="1,1"),
+	NrProcTestCase(origin="11,1",
+		single="1  1,1",
+		double="11,1",
+		triple="11,1"),
+	NrProcTestCase(origin="111,1",
+		single="1  1  1,1",
+		double="1  11,1",
+		triple="111,1"),
+	NrProcTestCase(origin="1111,1",
+		single="1  1  1  1,1",
+		double="11  11,1",
+		triple="1  111,1"),
+	NrProcTestCase(origin="1,11",
+		single="1,1  1",
+		double="1,11",
+		triple="1,11"),
+	NrProcTestCase(origin="1,111",
+		single="1,1  1  1",
+		double="1,1  11",
+		triple="1,111"),
+	NrProcTestCase(origin="1,1111",
+		single="1,1  1  1  1",
+		double="1,11  11",
+		triple="1,1  111"),
+	NrProcTestCase(origin="11,111",
+		single="1  1,1  1  1",
+		double="11,1  11",
+		triple="11,111"),
+	NrProcTestCase(origin="11,1111",
+		single="1  1,1  1  1  1",
+		double="11,11  11",
+		triple="11,1  111"),
+	NrProcTestCase(origin="111,111",
+		single="1  1  1,1  1  1",
+		double="1  11,1  11",
+		triple="111,111"),
+	NrProcTestCase(origin="1111,1111",
+		single="1  1  1  1,1  1  1  1",
+		double="11  11,11  11",
+		triple="1  111,1  111"),
+	# Decimal with letters
+	NrProcTestCase(origin="1a,234b",
+		single="1a,2  3  4b",
+		double="1a,2  34b",
+		triple="1a,234b"),
+	NrProcTestCase(origin="1a,2,345b",
+		single="1a,2,3  4  5b",
+		double="1a,2,3  45b",
+		triple="1a,2,345b"),
+	NrProcTestCase(origin="1a,2,3456b",
+		single="1a,2,3  4  5  6b",
+		double="1a,2,34  56b",
+		triple="1a,2,3  456b"),
+
+]
+
+thousandsSepDot = [
+	NrProcTestCase(origin="1.234",
+		single="1  2  3  4",
+		double="1  2  34",
+		triple="1  234"),
+	NrProcTestCase(origin="12.345",
+		single="1  2  3  4  5",
+		double="12  3  45",
+		triple="12  345"),
+	NrProcTestCase(origin="123.456",
+		single="1  2  3  4  5  6",
+		double="1  23  4  56",
+		triple="123  456"),
+	NrProcTestCase(origin="1.234.567",
+		single="1  2  3  4  5  6  7",
+		double="1  2  34  5  67",
```

Comment:
is this how people want this to speak with thousands seps? I'd assume it'd be 1 23 45 67

Reply:
Nope, 1,234,567 has two separators in it, and it has been discussed in #5930 that, if people put thousands separators in a number, they do it on purpose to group the number. Therefore, this code always respects the thousands separator as a separator for number pronunciation.

Reply:
@leonardder that doesn't really make sense to me.  Could you link to the comment where the discussion happened so I could understand that better. It doesn't make much sense to me to speak numbers in a grouped way on numbers with separators. For phone numbers, or credit card numbers it really makes sense. But when there is semantic meaning based on a digits position within a number, not so much. It would become very hard to get a concept of the scale of the number "one million two hundred and thirty four thousand five hundred and sixty seven" if it was read as "one, two, thirty four, five, sixty seven". For phone numbers and others, the separators are normally not related to the thousands separators, doing so would cause confusion about the meaning of the number.